### ISSUE TITLE
Camera preview showing gray rectangle

### ISSUE THREAD

**Issue by CaptainFlint** [qTox/qTox#5724]
##### Brief Description

OS: Windows 7 SP 1 (6.1) x64
qTox version: v1.16.3 x64
Commit hash: 8eed684c37bd711de2ca4bd863f5f0509edcdc2d
toxcore: 0.2.3
Qt: 5.9.6
Hardware: Logitech HD Pro C910

Reproducible: Always

Steps:
Start qTox, open settings, switch to Audio/Video.

Results:
My camera is selected in the list by default, so it automatically turns on (it has blue LEDs to indicate when it's working - they light up at this moment). The preview area initially shows a big grey-white avatar, then in a second or two it is replaced with a static gray rectangle, which remains there indefinitely. Camera activity LED remains lighted all the time, until I switch to another settings tab.

If I choose another resolution from the drop-down list, the preview turns black for a second, then goes back to grey. None of the resolution works, I tried them all.

The camera itself is working fine, I'm using it regularly in browser-based video call services.

Screenshot:
![image](https://user-images.githubusercontent.com/6475044/60048936-e672e280-96d5-11e9-85f2-b1118615e7db.png)

**Comment by anthonybilinski** [comment]
Hi @CaptainFlint, could you try a nightly build from here: https://github.com/qTox/qTox/blob/master/README.md and see if the problem persists? If so could you add debug logs from the nightly, which can be grabbed from settings -> Advanced -> copy debug log?

**Comment by CaptainFlint** [comment]
Yes, the problem is present in the nightly too (version: v1.16.3-476-g033f28e, commit hash: 033f28e67ecd61cd0fdb6cc0e8de8a2bc920e1d7).

Log: https://pastebin.com/jEZT602E

(I opened the tab, waited for the gray, then switched from 1080 to 720, waited for the gray, then copied the log.)

**Comment by sudden6** [comment]
Do you have the possibility to try a different webcam?

Does it work with a lower resolution?

I suspect the camera might use a codec qTox doesn't detect, but I'm not sure if that could result in the gray images.

**Comment by anthonybilinski** [comment]
From the log, CaptainFlint's getting `pixel format: unknown`, possibly we don't have support for the pixel format http://wiki.oz9aec.net/index.php/Logitech_HD_Pro_Webcam_C910 uses in our windows directshow?

**Comment by sudden6** [comment]
The pixel formats listed there for CaptainFlint's resolution and FPS setting only allow for `MJPG` pixel format and I think we support that. IIRC the pixel format is only printed on Linux.

**Comment by CaptainFlint** [comment]
> Do you have the possibility to try a different webcam?

Unfortunately, no, I don't have any other webcam I could plug in.

Although I do have a laptop with an integrated webcam, so I checked the same nightly there, and the preview worked fine. The log is: https://pastebin.com/W3XeK7ik OS on this laptop is Windows 10 x64.

Unfortunately I have no idea what manufacturer produced this camera, there's no proper info in the device manager. Googling for VID 04F2, PID B5AB does not help much; the drivers are produced by SunplusIT, but I've never heard of them.

> Does it work with a lower resolution?

No. There are the following resolutions available: 120p, 288p, 480p, 720p, 1080p. I've checked them all, and they all give gray rectangle.

**Comment by Xayaan** [comment]
@CaptainFlint For god sakes man, It's 2019 and who uses Windows 7?????? This issue is obsolete.

**Comment by CaptainFlint** [comment]
`who uses Windows 7?`
Lots of people do.

`This issue is obsolete.`
No, it's not.
